src/Google.Bigquery.V2/BigqueryClientImpl.Queries.cs

```diff
@@ -27,47 +29,64 @@ public partial class BigqueryClientImpl
 
-            var queryResponse = Service.Jobs.Query(new Apis.Bigquery.v2.Data.QueryRequest { Query = sql }, ProjectId).Execute();
+            var queryRequest = new Apis.Bigquery.v2.Data.QueryRequest { Query = sql };
+            options?.ModifyRequest(queryRequest);
+            var request = Service.Jobs.Query(queryRequest, ProjectId);
+            var queryResponse = request.Execute();
             return new BigqueryResult(this, queryResponse);
         }
 
         /// <inheritdoc />
-        public override BigqueryJob CreateQueryJob(string sql, TableReference destinationTable = null)
+        public override BigqueryJob CreateQueryJob(string sql, CreateQueryJobOptions options = null)
         {
             Preconditions.CheckNotNull(sql, nameof(sql));
-
+            var query = new JobConfigurationQuery { Query = sql };
+            options?.ModifyRequest(query);
             var job = Service.Jobs.Insert(new Job
             {
                 Configuration = new JobConfiguration
                 {
-                    Query = new JobConfigurationQuery
-                    {
-                        Query = sql,
-                        DestinationTable = destinationTable
-                    },
+                    Query = query
                 },
             }, ProjectId).Execute();
             return new BigqueryJob(this, job);
         }
 
         /// <inheritdoc />
-        public override BigqueryResult GetQueryResult(JobReference jobReference)
+        public override BigqueryResult GetQueryResults(JobReference jobReference, GetQueryResultsOptions options = null)
         {
             Preconditions.CheckNotNull(jobReference, nameof(jobReference));
 
-            var firstResponse = Service.Jobs.GetQueryResults(jobReference.ProjectId, jobReference.JobId).Execute();
-            return new BigqueryResult(this, firstResponse, () => Service.Jobs.GetQueryResults(jobReference.ProjectId, jobReference.JobId));
+            Func<GetQueryResultsRequest> requestProvider = () =>
+            {
+                var request = Service.Jobs.GetQueryResults(jobReference.ProjectId, jobReference.JobId);
+                options?.ModifyRequest(request);
+                return request;
+            };
+            var firstResponse = requestProvider().Execute();
+            return new BigqueryResult(this, firstResponse, requestProvider);
         }
 
         /// <inheritdoc />
-        public override BigqueryResult ListRows(TableReference tableReference, TableSchema schema = null)
+        public override BigqueryResult ListRows(TableReference tableReference, TableSchema schema = null, ListRowsOptions options = null)
         {
             Preconditions.CheckNotNull(tableReference, nameof(tableReference));
             schema = schema ?? GetSchema(tableReference);
+            
+            var projectId = tableReference.ProjectId;
+            var datasetId = tableReference.DatasetId;
+            var tableId = tableReference.TableId;
```

Comment:
Inconsistent dereferencing location as compared to GetQueryResults. Doesn't really matter, but might be tidier to make them conform?